src/main/java/org/example/AppLauncher.java

```diff
@@ -1,8 +1,27 @@
 package org.example;
 
+import org.example.sum.*;
+import org.example.sum.NumberGetter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class AppLauncher {
+    private static final Logger logger = LoggerFactory.getLogger(AppLauncher.class);
+
     public static void main(String[] args) {
-        SumCalculator sumCalculator = new SumCalculator();
-        System.out.println("Sum of numbers: " + sumCalculator.sum(7));
+        try {
+            NumberGetter userInput = new UserInput();
+            NumberManager numberManager = new NumberManager(userInput);
+//        NumberGetter randomInt = new RandomInt();
+//        NumberManager numberManager = new NumberManager(randomInt);
+//            NumberGetter numberFromFile = NumberFromFile.getInstance("src/main/resources/number.txt");
+//            NumberManager numberManager = new NumberManager(numberFromFile);
+            int number = numberManager.getNumericValue();
+            logger.info("Retrieved number: {}", number);
+            SumCalculator sumCalculator = new SumCalculator();
+            logger.info("Sum of numbers from 1 to {}: {}", number, sumCalculator.sum(number));
+        } catch (IllegalStateException | IllegalArgumentException e) {
```

Comment:
> Issue: The IllegalStateException and IllegalArgumentException are caught together. However, it might be beneficial to handle them separately if different actions are required.
Solution: Separate the exception handling blocks if different actions should be taken based on the exception type.

```
try {
    // existing code...
} catch (IllegalStateException e) {
    logger.error("Illegal state encountered: {}", e.getMessage(), e);
} catch (IllegalArgumentException e) {
    logger.error("Invalid argument provided: {}", e.getMessage(), e);
}
```

Reply:
My only goal is to log the exception with the same message for both `IllegalStateException` and `IllegalArgumentException` and I don't plan to handle them separately